framework/assets/yii.validation.js

```diff
@@ -244,7 +244,15 @@ yii.validation = (function ($) {
 
             value = $input.val();
             if (!options.skipOnEmpty || !pub.isEmpty(value)) {
-                value = $.trim(value);
+                value = new String(value);
```

Comment:
If this implementation is borrowed from somewhere, a link will be useful.

Reply:
code based on https://javascript.ru/php/trim

Reply:
Got it. Maybe leave it as a comment?

Reply:
@arogachev top code is invalid, it fixed in comment

Reply:
Actually maybe use jQuery implementation? What it looks like? The benefit is it's already tested and more predictable because it was already used. :thinking:

Reply:
@arogachev 
> Note: This API has been deprecated in jQuery 3.5; please use the native String.prototype.trim method instead.

or you mean something other?